addons/point_of_sale/tests/test_pos_products_with_tax.py

```diff
@@ -0,0 +1,348 @@
+            payments=[(self.cash_pm, 104.01)],
+            customer=self.customer,
+            is_invoiced=True,
+            uid='12345-123-1234',
+        ))
+
+        # sync orders
+        order = self.env['pos.order'].create_from_ui(orders)
+
+        # check values before closing the session
+        self.assertEqual(1, self.session.order_count)
+        self.assertEqual(1, self.session.payment_count)
+
+        orders_total = sum(order.amount_total for order in self.session.order_ids)
+        payments_total = sum(payment.amount for payment in self.session.order_ids.mapped('payment_ids'))
+        self.assertAlmostEqual(orders_total, payments_total, msg='Total order amount should be equal to the total payment amount.')
+
+        # return order
+        order_to_return = self.session.order_ids.filtered(lambda order: '12345-123-1234' in order.pos_reference)
+        order_to_return.refund()
+
+        refund_order = self.session.order_ids.filtered(lambda order: order.state == 'draft')
+        context_make_payment = {"active_ids": [refund_order.id], "active_id": refund_order.id}
+        make_payment = self.env['pos.make.payment'].with_context(context_make_payment).create({
+            'payment_method_id': self.cash_pm.id,
+            'amount': -104.01,
+        })
+        make_payment.check()
+        self.assertEqual(refund_order.state, 'paid', 'Payment is registered, order should be paid.')
+        self.assertAlmostEqual(refund_order.amount_paid, -104.01, msg='Amount paid for return order should be negative.')
+
+        # close the session
+        self.session.action_pos_session_validate()
+
+        # check values after the session is closed
+        session_move = self.session.move_id
+
+        # instead of credit, the sales line should be debit
+        sales_lines = session_move.line_ids.filtered(lambda line: line.account_id == self.sale_account)
+        self.assertAlmostEqual(sum(sales_lines.mapped('balance')), 93.63)
+
+        receivable_line_bank = session_move.line_ids.filtered(lambda line: self.bank_pm.name in line.name)
+        self.assertFalse(receivable_line_bank, msg='There should be no bank receivable line.')
+
+        receivable_line_cash = session_move.line_ids.filtered(lambda line: self.cash_pm.name in line.name)
+        self.assertFalse(receivable_line_cash, msg='There should be no cash receivable line.')
+
+        manually_calculated_taxes = (4.01, 6.37)  # should be positive since it is return order
+        tax_lines = session_move.line_ids.filtered(lambda line: line.account_id == self.tax_received_account)
+        self.assertAlmostEqual(sum(manually_calculated_taxes), sum(tax_lines.mapped('balance')))
+        for t1, t2 in zip(sorted(manually_calculated_taxes), sorted(tax_lines.mapped('balance'))):
+            self.assertAlmostEqual(t1, t2, msg='Taxes should be correctly combined and should be debit.')
+
+    def test_picking(self):
```

Comment:
I think this test is useless because your products are not of type 'product', but type 'consu', so it'll always work. And this test is only testing that validating a picking works, which is already done in the stock tests.

You can remove it and, just add a line in another of your test checking that there is a picking on an order, and state == 'done'